### PR TITLE
47132 vaos va link/requested appointments

### PR DESCRIPTION
**Pull request by ryanshaw**
## Description
This is one of several PRs that replace the react-router `Link` component in VAOS with the newer `va-link` web component.

[va-link](https://design.va.gov/components/link/)
[va-link storybook](https://design.va.gov/storybook/?path=/docs/components-va-link--default)

This was broken up into multiple PR's to allow for easier code review as this change impacts nearly every file and every test within VAOS.

## Original issue(s)
[department-of-veterans-affairs/va.gov-team/issues/47132](https://github.com/department-of-veterans-affairs/va.gov-team/issues/47132)

## Related to these other PR's:
- [breadcrumbs](https://github.com/department-of-veterans-affairs/vets-website/pull/22779)
- [covid flow](https://github.com/department-of-veterans-affairs/vets-website/pull/22818)
- [new appointment](https://github.com/department-of-veterans-affairs/vets-website/pull/23125)
- [appointment list](https://github.com/department-of-veterans-affairs/vets-website/pull/23128)

## Testing done
- Unit testing
- e2e testing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx [hunk 1]
```diff
@@ -59,17 +59,20 @@ export default function RequestedStatusAlert({ appointment, facility }) {
             available appointment.
             <br />
             <div className=" vads-u-margin-top--1">
-              <Link to="/" onClick={handleViewClick}>
-                View your appointments
-              </Link>
+              <va-link
+                href="/"
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx:63]: This link is going to the wrong path. "/" was enrolled in a special routes file. I don't know how it works but @vbahinwillit may have the background on it.

File: src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx [hunk 2]
```diff
@@ -59,17 +59,20 @@ export default function RequestedStatusAlert({ appointment, facility }) {
...
+                onClick={handleViewClick}
+                text="View your appointments"
+                data-testid="view-appointments-link"
+              />
             </div>
             <div className=" vads-u-margin-top--1">
-              <Link
-                to="/new-appointment"
+              <va-link
+                href="/new-appointment"
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx:71]: This link is going to the wrong path. "/new-appointment" was enrolled in a special routes file. I don't know how it works but @vbahinwillit may have the background on it.

**Review by jenniemc — CHANGES_REQUESTED** [review]
The details link is working perfectly but the links in the status alert is going to the wrong path.

File: src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx (hunk 2 above)
Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx:71]: @jenniemc I also need to look at why tests were passing, they should not have been. I may need to adjust those.

**Comment by diondredroidios** [comment]
please text me 704 363 1339

u.s. citizen

available now

20 years experience

max rate

available immediately

1099/w2/c2c with preferences for FTE + signing bonus

in Charlotte NC

4615 last 4

28202

11/25 DOB

interview times 10-3:00 PM EST

just rolled off contract, 5/2022 no lead time needed

highest level of education: bachelors duke 2000 comp sci

vaccinated yes

preferred remote



Please text me 704 363 1339, that's the fastest way to get ahold of me.

SEND RTR for max rate

20 years experience

9 years node.js

Kotlin SME

8 years react / react native

10 years android / iOS

6 years docker/Kubernetes

10 years devOps/CI/CD

10 years xamarin

8 years AWS

10 years Senior/lead

7 years blockchain

8 years dev lead

12 years javascript

6 years typescript

8 years cloud

8 years hands-on team lead

20 years experience

u.s. citizen

available now

max rate

available immediately

1099/w2/c2c with preferences for FTE + signing bonus

in Charlotte NC

4615 last 4 sen

28202

please send RTR

11/25 DOB

interview times 10-3:00 PM EST

Github diondredroidios


Job descriptions:

1. Ijust rolled off contract, 8/2022 no lead time needed for onboarding new
job

2. I functioned as hands-on development lead at Meta/Facebook, doing 80%
heads down coding and 20% leadership activities

3. I was a lead of 8 engineers, working on the FB Messenger tool,
React/Angular/React Native, Node.js , Postgres, C#, IOS/Android,
Jenkins/Bitrise

4. Functioned as hands-on senior developer/dev lead for 8 years.

5. I can function as a fully-functional top senior individual contributor
developer or hands-on developer lead for startup to Fortune 500 companies
and I have no preference.


Education

1. highest level of education: bachelors duke 2000 comp sci

2. vaccinated yes

3. preferred remote

File: src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx (hunk 1 above)
Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx]: > The details link is working perfectly but the links in the status alert is going to the wrong path.

@jenniemc I changed this to use the va-link component with the React bindings to handle the onclick event.

**Review by vbahinwillit — APPROVED** [review]
LGTM

**Review by simiadebowale — APPROVED** [review]
LGTM

**Review by jenniemc — APPROVED** [review]
This is terrific, the links are working as expected! Just have some minor suggestions.

File: src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx [hunk 3]
```diff
@@ -59,17 +56,23 @@ export default function RequestedStatusAlert({ appointment, facility }) {
             available appointment.
             <br />
             <div className=" vads-u-margin-top--1">
-              <Link to="/" onClick={handleViewClick}>
-                View your appointments
-              </Link>
+              <VaLink
```
[the remark this answers is not in the document]
Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx]: Made this change.

File: src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx [hunk 4]
```diff
@@ -59,17 +56,23 @@ export default function RequestedStatusAlert({ appointment, facility }) {
...
+              />
             </div>
             <div className=" vads-u-margin-top--1">
-              <Link
-                to="/new-appointment"
-                onClick={handleNewApptClick(dispatch)}
-              >
-                New appointment
-              </Link>
+              <VaLink
```
[the remark this answers is not in the document]
Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/RequestedStatusAlert.jsx]: Made this change...

**Pull request merged**